README.md

```diff
@@ -2,24 +2,250 @@
+
+#### "Classes" of Standard
+The PMIx Standard is composed of a combination of APIs and key-value attributes. APIs are intentionally designed to be "lightweight" - i.e., the API signature is generic with a minimal number of fixed parameters in its signature. With few exceptions, APIs are required to include an array of key-value attributes in their signature to facilitate behavioral extensions without modifying the API signature. Thus, PMIx attempts to minimize deprecation and modification of APIs as use models evolve over time.
+
+Each API can designate a given attribute as either "**required**" or "**optional**" - this indicates whether or not a PMIx implementation of that API is required to support the attribute. PMIx implementations are requested to support optional attributes, but are not required to do so. This is intended to both provide flexibility to PMIx implementors (e.g., to tailor the implementation to their target audience) and to ease adoption of the Standard.
+
+Attributes can similarly be designated by the caller of a PMIx API as either "required" or "optional". This serves as a directive to the PMIx implementation as to the expected response should the implementation not support that attribute. Attributes provided by the caller that are marked as "required" but are not supported by the API must result in the API returning "not supported". Provided attributes marked as "optional" may be ignored if not supported.
+
+Both PMIx APIs and attributes are further grouped into three classes:
+
+* "**Provisional**"" entries in the Standard are officially part of the Standard but are not yet guaranteed to be stable - e.g., an API signature and/or attributes it supports may be changed, or the entry can be completely removed if subsequent evaluation proves negative. New provisional entries can be added to any minor release of the Standard. Subsequent changes to an entry requires warning in at least one minor release of the Standard document before acceptance - i.e., a proposed change to a provisional entry announced in version A.1 of the PMIx Standard may be committed in version A.2 of the Standard. This includes deprecation of an existing provisional entry. Thus, the intent is to provide a relatively easy mechanism by which proposals acceptable in principle to the community can enter the Standard prior to being fully stabilized, thereby allowing early adopters an opportunity to explore the proposed capabilities.
+
+    All proposed additions to the PMIx Standard enter the PMIx Standard under Provisional status. Attributes in the "Provisional" class may only be designated as "optional" by an API - i.e., an API definition cannot designate a "Provisional" attribute as "required". Users, however, are always free to "require" that an attribute be supported when calling the API, thus directing the API's response should the attribute not be supported by it.
```

Comment:
I think I understand what you are going for here, but it is a bit confusing since there is so much flexibility in what should be implemented in the standard already. What if the provisional addition is meant to be a required interface/attribute in the stable version in order for some functionality to work? How can one try it out properly without knowing the required/optional parts of the eventual stable interface?

Reply:
> Attributes that are intended to become designated as "required" by an API when elevated to "Stable" status shall be clearly designated as such in the proposal.